gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TripEdgeArrivalDepartureTimeValidatorTest.java

```diff
@@ -0,0 +1,262 @@
+          .setTripId("trip1")
+          .setStopId("1003")
+          .setStopSequence(3)
+          .setArrivalTime(GtfsTime.fromString("12:52:34"))
+          .setDepartureTime(GtfsTime.fromString("12:53:34"))
+          .build();
+  private final GtfsStopTime endStopTimeHasArrival =
+      new GtfsStopTime.Builder()
+          .setCsvRowNumber(23)
+          .setTripId("trip1")
+          .setStopId("1003")
+          .setStopSequence(3)
+          .setArrivalTime(GtfsTime.fromString("12:52:34"))
+          .build();
+  private final GtfsStopTime endStopTimeHasDeparture =
+      new GtfsStopTime.Builder()
+          .setCsvRowNumber(23)
+          .setTripId("trip2")
+          .setStopId("1003")
+          .setStopSequence(3)
+          .setDepartureTime(GtfsTime.fromString("12:53:34"))
+          .build();
+  private final GtfsStopTime endStopTimeHasNeither = new GtfsStopTime.Builder()
+                                                         .setCsvRowNumber(23)
+                                                         .setTripId("trip2")
+                                                         .setStopId("1003")
+                                                         .setStopSequence(3)
+                                                         .build();
+  private final GtfsStopTime middleStop1 = new GtfsStopTime.Builder()
+                                               .setCsvRowNumber(41)
+                                               .setTripId("trip1")
+                                               .setStopId("1002")
+                                               .setStopSequence(2)
+                                               .build();
+  private final GtfsStopTime middleStop2 = new GtfsStopTime.Builder()
+                                               .setCsvRowNumber(41)
+                                               .setTripId("trip2")
+                                               .setStopId("1002")
+                                               .setStopSequence(2)
+                                               .build();
+  @Test
+  public void validShouldNotGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    TripEdgeArrivalDepartureTimeValidator validator = new TripEdgeArrivalDepartureTimeValidator();
+
+    // Create stopTimeTable:
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(startStopTimeHasBoth);
+    stopTimes.add(middleStop1);
+    stopTimes.add(endStopTimeHasBoth);
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+
+    // Create tripTable:
+    List<GtfsTrip> trips = new ArrayList<>();
+    trips.add(trip1);
+    validator.tripTable = GtfsTripTableContainer.forEntities(trips, noticeContainer);
+
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices()).isEmpty();
+  }
+
+  @Test
+  public void missingDepartureTimesForStartAndEndStop() {
```

Comment:
I would lean towards having individual tests for each version of the notice, e.g. `missingDepartureTimeForStartStop`, `missingDepartureTimeForEndStop`.

Reply:
Ok, updated now with 9 unit tests